### ISSUE TITLE
BookStore-Modular: NU1105

### ISSUE THREAD

**Issue by leonkosak** [abpframework/abp#2654]
I made exactly the same structure ("application" and "modules" folders) and crate one application and one module.
I started adding reference in application (references to modules csproj files) .csproj file (<ProjectReference>...) and then "typeof(...) i ...Module.cs), but I cannot compile because of NU1105.

Example message:
Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1105	Unable to find project information for 'C:\_Demo\abpCom200\Demo200Modular\modules\ModuleDatabase\src\ModuleDatabase.Application.Contracts\ModuleDatabase.Application.Contracts.csproj'. If you are using Visual Studio, this may be because the project is unloaded or not part of the current solution so please run a restore from the command-line. Otherwise, the project file may be invalid or missing targets required for restore.	Demo200Modular.Application.Contracts	C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Application.Contracts\Demo200Modular.Application.Contracts.csproj	1	

There is no yellow sign at module project (so the path in application .csproj project) is correct.

What should I do?

**Comment by ismcagdas** [comment]
@leonkosak could you firs build the module project and than run `dotnet restore` for your main application ?

**Comment by leonkosak** [comment]
![image](https://user-images.githubusercontent.com/39346964/72517854-661ff480-3854-11ea-8c92-958df718018d.png)

@ismcagdas I go that order of adding project references od modules scproj in application csproj projects.

Tha problem starts from adding .EntityFrameworkCore project in application project. In AppNameModularEntityFrameworkCoreModule.cs, the ModuleNameEntityFrameworkCoreModule cannot be found (when I tried to add in DependsOn section.

The same problem is present for .HttpApi, .HttpApi.Client.

P.S.: Application is abp Commercial (2.0.0.) and module was generated on the web site abp.io.

**Comment by leonkosak** [comment]
I tried one more time from scratch and the same issues appeared.

**Comment by leonkosak** [comment]
@ismcagdas The most interesting thing to me is that in .EntityFrameworkCore, .HttpApi, .HttpApi.Client projects, the  xxxModule (for Module) cannot be found).
I tried changing to netstandard2.1 in csproj in module projects, but this changes nothing.

**Comment by leonkosak** [comment]
dotnet restore log for application (for .sln).

PS C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core> dotnet restore
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.HttpApi.Client\Demo200Modular.HttpApi.Client.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Application\Demo200Modular.Application.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Domain.Shared\Demo200Modular.Domain.Shared.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Domain\Demo200Modular.Domain.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Application.Contracts\Demo200Modular.Application.Contracts.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 122.11 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Domain.Shared\Demo200Modular.Domain.Shared.csproj.
  Restore completed in 122.25 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.HttpApi.Client\Demo200Modular.HttpApi.Client.csproj.
  Restore completed in 122.1 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Application.Contracts\Demo200Modular.Application.Contracts.csproj.
  Restore completed in 122.09 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Domain\Demo200Modular.Domain.csproj.
  Restore completed in 122.24 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.Application\Demo200Modular.Application.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.HttpApi\Demo200Modular.HttpApi.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 12.9 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.HttpApi\Demo200Modular.HttpApi.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.HttpApi.Client.ConsoleTestApp\Demo200Modular.HttpApi.Client.ConsoleTestApp.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 65.69 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.HttpApi.Client.ConsoleTestApp\Demo200Modular.HttpApi.Client.ConsoleTestApp.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.TestBase\Demo200Modular.TestBase.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 45.52 ms for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.TestBase\Demo200Modular.TestBase.csproj.
  Restore completed in 9.97 ms for C:\_Demo\abpCom200\Demo200Modular\modules\ModuleDatabase\src\ModuleDatabase.Application.Contracts\ModuleDatabase.Application.Contracts.csproj.
  Restore completed in 10.03 ms for C:\_Demo\abpCom200\Demo200Modular\modules\ModuleDatabase\src\ModuleDatabase.Application\ModuleDatabase.Application.csproj.
  Restore completed in 10.39 ms for C:\_Demo\abpCom200\Demo200Modular\modules\ModuleDatabase\src\ModuleDatabase.Domain.Shared\ModuleDatabase.Domain.Shared.csproj.
  Restore completed in 10.19 ms for C:\_Demo\abpCom200\Demo200Modular\modules\ModuleDatabase\src\ModuleDatabase.Domain\ModuleDatabase.Domain.csproj.
  Restore completed in 20.61 ms for C:\_Demo\abpCom200\Demo200Modular\modules\ModuleDatabase\src\ModuleDatabase.EntityFrameworkCore\ModuleDatabase.EntityFrameworkCore.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.EntityFrameworkCore\Demo200Modular.EntityFrameworkCore.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.EntityFrameworkCore.DbMigrations\Demo200Modular.EntityFrameworkCore.DbMigrations.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 4.57 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.EntityFrameworkCore.DbMigrations\Demo200Modular.EntityFrameworkCore.DbMigrations.csproj.
  Restore completed in 4.57 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.EntityFrameworkCore\Demo200Modular.EntityFrameworkCore.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.DbMigrator\Demo200Modular.DbMigrator.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.EntityFrameworkCore.Tests\Demo200Modular.EntityFrameworkCore.Tests.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.Domain.Tests\Demo200Modular.Domain.Tests.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 5.21 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.DbMigrator\Demo200Modular.DbMigrator.csproj.
  Restore completed in 5.08 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.EntityFrameworkCore.Tests\Demo200Modular.EntityFrameworkCore.Tests.csproj.
  Restore completed in 5.16 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.Domain.Tests\Demo200Modular.Domain.Tests.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.Application.Tests\Demo200Modular.Application.Tests.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 7.4 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\test\Demo200Modular.Application.Tests\Demo200Modular.Application.Tests.csproj.
C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.HttpApi.Host\Demo200Modular.HttpApi.Host.csproj : warning NU1701: Package 'BouncyCastle 1.8.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\Demo200Modular.sln]
  Restore completed in 8.42 sec for C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core\src\Demo200Modular.HttpApi.Host\Demo200Modular.HttpApi.Host.csproj.
PS C:\_Demo\abpCom200\Demo200Modular\application\aspnet-core> C:\_Demo\abpCom200\Demo200Modular\application\aspnet-coreC:\_Demo\abpCom200\Demo200Modular\application\aspnet-core

**Comment by leonkosak** [comment]
@ismcagdas any updates on this issue?

**Comment by ismcagdas** [comment]
@leonkosak could you share the project via email ? I can reproduce the problem and assist you faster like that.

**Comment by ismcagdas** [comment]
Thank you for sharing the project @leonkosak. It seems like you are missing a using directive in Demo200ModularEntityFrameworkCoreModule.cs. After adding below using statement project builds successfully.

`using ModuleDatabase.EntityFrameworkCore;`

**Comment by leonkosak** [comment]
I tried that in the past, but .EntityFrameworkCore is not found and that's confusing me.

![image](https://user-images.githubusercontent.com/39346964/72611613-db5df900-392a-11ea-9bae-8e1a4e30cd11.png)

The same problem is in .HttpApi and .HttpApi.Client

In the other projects, when I added using ModuleDatabase, there were no issues at all adding Module in DependsOn section.

Am I missing something to re-confogure in Visual Studio?

**Comment by ismcagdas** [comment]
@leonkosak here is what I did before opening VS at all.

1. Go to "modules\ModuleDatabase" folder and run `dotnet restore` and `dotnet build`.
2. Go to "application\aspnet-core" folder and run `dotnet restore` and `dotnet build`.

After step 2, I got a build error. Than I added the using statement and runned `dotnet restore` and `dotnet build` commands in "application\aspnet-core" and it worked.

**Comment by leonkosak** [comment]
@ismcagdas Thank you. It works. But I can understand why in the first projects in application (.Application, .Application.Contracts, .Domain, .Domain.Shared worked without that?

**Comment by kurtwilbies** [comment]
@leonkosak Can you share your solution?

**Comment by leonkosak** [comment]
It's really nothing special (I currently have no access to this demo project).
Just "application" and "modules" folders. In application folder, there is abp Commercial.
In modules folder, there is just one module. In abp Commercial are references to this module ([like this](https://github.com/abpframework/abp/issues/2654#issuecomment-575091535))

I succeassfully added one method on API in module, compile everything and when the application (abp Commercial) API is launched (Swagger), there is method from module's API and it works.

That's it.

I don't know how to achive that I define SQL database (tables) in module and then run migrations "on application level - abp Commercial) so that all SQL tables (C# classes which are in DbSet) would be merged into single database (and each tenant would have the same structure when new tenant would be created on application level).

**Comment by ismcagdas** [comment]
@leonkosak you need to add migrations like this https://github.com/abpframework/abp/blob/dev/templates/app/aspnet-core/src/MyCompanyName.MyProjectName.EntityFrameworkCore.DbMigrations/EntityFrameworkCore/MyProjectNameMigrationsDbContext.cs#L41. I assume your module has a similar extension method like this https://github.com/abpframework/abp/blob/dev/modules/tenant-management/src/Volo.Abp.TenantManagement.EntityFrameworkCore/Volo/Abp/TenantManagement/EntityFrameworkCore/AbpTenantManagementDbContextModelCreatingExtensions.cs#L10
